compiler/util/misc.cpp

```diff
@@ -246,7 +246,7 @@ static void print_user_internal_error() {
 
   error[idx++] = '-';
   // next 4 characters are the line number
-  sprintf(&error[idx], "%04d", err_lineno);
+  snprintf(&error[idx], 4 * sizeof(char), "%04d", err_lineno);
```

Comment:
`snprintf` docs say:

> If the resulting string would be longer than n-1 characters, the remaining characters are discarded and not stored, but counted for the value returned by the function.

Because one of the characters is used for a null terminator. I'm assuming `%04d` is "print a number padded to the left up to 4 characters", which means you'll get chomped for one character. I think you need to bump your sizeof to 5.

Reply:
Good catch!